src/library_nodefs.js

```diff
@@ -78,18 +83,18 @@ mergeInto(LibraryManager.library, {
     // This maps the integer permission modes from http://linux.die.net/man/3/open
     // to node.js-specific file open permission strings at http://nodejs.org/api/fs.html#fs_fs_open_path_flags_mode_callback
     flagsForNode: function(flags) {
-      flags &= ~0x200000 /*O_PATH*/; // Ignore this flag from musl, otherwise node.js fails to open the file.
-      flags &= ~0x800 /*O_NONBLOCK*/; // Ignore this flag from musl, otherwise node.js fails to open the file.
-      flags &= ~0x8000 /*O_LARGEFILE*/; // Ignore this flag from musl, otherwise node.js fails to open the file.
-      flags &= ~0x80000 /*O_CLOEXEC*/; // Some applications may pass it; it makes no sense for a single process.
+      flags &= ~{{{ cDefine('O_PATH') }}}; // Ignore this flag from musl, otherwise node.js fails to open the file.
+      flags &= ~{{{ cDefine('O_NONBLOCK') }}}; // Ignore this flag from musl, otherwise node.js fails to open the file.
+      flags &= ~{{{ cDefine('O_LARGEFILE') }}}; // Ignore this flag from musl, otherwise node.js fails to open the file.
+      flags &= ~{{{ cDefine('O_CLOEXEC') }}}; // Some applications may pass it; it makes no sense for a single process.
+      flags &= ~{{{ cDefine('O_DIRECTORY') }}}; // Node.js doesn't need this passed in, it errors.
```

Comment:
How about we take this part, along with the struct_info.json change and make a new change out of it?  

It does look like have a couple of tests that use `O_DIRECTORY`:

- tests/fs/test_getdents64.cpp
- tests/fs/test_getdents64_special_cases.cpp

We could enable them under node as part of that separate change?

Reply:
Or is this part needed to support the use of `O_DIRECTORY` in musl/src/dirent/opendir.c maybe?

Reply:
Yeah, looks like the check was moved to open() and here needs to be stripped so it isn't passed to node — so it's necessary for the opendir fix, but it could be combined with the open() change and could be landed first, before the rest of the changes?

Reply:
I'll go back to fixing tests for now, happy to break this up however makes sense though.